zeroconf/src/linux/service.rs

```diff
@@ -115,6 +132,32 @@ impl AvahiServiceContext {
             name: None,
             kind: c_string!(kind),
             port,
+            subtypes: None,
+            group: None,
+            txt_record: None,
+            interface_index: avahi_sys::AVAHI_IF_UNSPEC,
+            domain: None,
+            host: None,
+            registered_callback: None,
+            user_context: None,
+        }
+    }
+
+    fn with_sub_types(kind: &str, port: u16, sub_types: Vec<&str>) -> Self {
+        Self {
+            client: None,
+            name: None,
+            kind: c_string!(kind),
+            port,
+            subtypes: Some(
+                sub_types
+                    .iter()
+                    .map(|sub_type| {
+                        let sub_type = format!("_{sub_type}._sub.{kind}");
```

Comment:
Is this enforcement of the sub-type following this format required? Do we have any documentation that details this spec?

Reply:
I don't see an indication that a subtype must follow this, but it is the format used throughout Section 7.1 of RFC6763: https://datatracker.ietf.org/doc/html/rfc6763#page-21

Reply:
Actually, here's the section, lower down: 

```
A fully qualified domain name may be up to 255 bytes long, plus one
byte for the final terminating root label at the end.  Domain names
used by DNS-SD take the following forms:

                   <sn>._tcp . <servicedomain> . <parentdomain>.
      <Instance> . <sn>._tcp . <servicedomain> . <parentdomain>.
      <sub>._sub . <sn>._tcp . <servicedomain> . <parentdomain>.
```

And below that:
```
When using subtypes, the subtype identifier is allowed to be up to 63
bytes, plus the length byte, making 64.  Including the "_sub" and its
length byte, this makes 69 bytes.
```

Reply:
Thank you providing this.